engine/crates/engine/src/request.rs

```diff
@@ -34,19 +34,38 @@ pub struct Request {
 
     /// The extensions config of the request.
     #[serde(default)]
-    pub extensions: HashMap<String, Value>,
+    pub extensions: RequestExtensions,
+}
+
+#[derive(Debug, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct RequestExtensions {
+    #[serde(default)]
+    pub persisted_query: Option<PersistedQueryRequestExtension>,
+    #[serde(flatten)]
+    pub custom: HashMap<String, Value>,
+}
+
+#[serde_with::serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PersistedQueryRequestExtension {
+    pub version: u32,
+    #[serde_as(as = "serde_with::hex::Hex")]
+    pub sha256_hash: Vec<u8>,
```

Comment:
Could have encoded this as a versioned enum.  No big deal though

Reply:
I tried, but versioned enum only work with strings, not integers :/
https://github.com/serde-rs/serde/issues/745

Reply:
could make a custom Deserialize, but felt way overkill :D

Reply:
Ah ok, never mind then